### ISSUE TITLE
Service Name is displayed as otlpresourcenoattributes when using OtlpExporter, opentelemetry-collector and zipkin

### ISSUE THREAD

**Issue by dgrozenok** [open-telemetry/opentelemetry-dotnet#1294]
# Bug Report

NuGet Packages:
- OpenTelemetry 0.6.0-beta.1
- OpenTelemetry.Exporter.OpenTelemetryProtocol 0.6.0-beta.1
- OpenTelemetry.Extensions.Hosting 0.6.0-beta.1
- OpenTelemetry.Instrumentation.AspNetCore 0.6.0-beta.1
- OpenTelemetry.Instrumentation.Http 0.6.0-beta.1

Runtime version netcoreapp3.1

ASPNET.Core application with the AspNetCoreInstrumentation, HttpClientInstrumentation and OtlpExporter sending the traces to the opentelemetry-collector v 0.10.0 that in its turn sends them to zipkin.

Expected result is the names of the services endpoints to appear in zipkin.

Actual result is all the spans appear in zipkin as OTLPRESOURCENOATTRIBUTES.

**Comment by eddynaka** [comment]
@dgrozenok , can you show us how are you initializing the sdk?

**Comment by dgrozenok** [comment]
Created a sample project to reproduce here https://github.com/dgrozenok/OtlpDemo

The initialization is here 
https://github.com/dgrozenok/OtlpDemo/blob/a162510cfc10a7d26a874a6e38dc3a6bc4397880/OtlpDemo/Startup.cs#L25

I realize that if I uncomment the line

// .SetResource(new Resource(new[] {new KeyValuePair<string, object>("service.name", "WebApi")}))

all the spans will have WebApi name. However, Zipkin exporter is doing much better job of mapping different instrumentation based attributes like http.host to the ServerAddress making the spans looking more meaningful.

**Comment by dgrozenok** [comment]
One suggestion may be to making the ServiceName as part of the Otlp Exporter options similar to the Zipkin exporter to avoid the .SetResource() call with the hard-coded key "service.name".

The issue with mapping the attributes produced by the instrumentation libraries may actually be on the opentelemetry-collector level. Not sure which attributes are part of the Otlp protocol and where the mapping is supposed to happen by design. It may be an issue with the instrumentation libraries as well. Just would like to have the chain Instrumentation->OtlpExporter->Otel-Collector->Zipkin to work with the same result as Instrumentation->ZipkinExporter->Zipkin.

**Comment by CodeBlanch** [comment]
Both Jaeger & Zipkin exporters have logic to resolve remote service name based on other attributes that might be present:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerActivityExtensions.cs#L44

https://github.com/open-telemetry/opentelemetry-dotnet/blob/a5e7d6e631c1b5b9b15c4bbcfbb50deaf55cfc26/src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinActivityConversionExtensions.cs#L33

I haven't applied this logic to OtlpExporter yet. I can take a look at doing that if no one picks this up.

**Comment by CodeBlanch** [comment]
@dgrozenok I tested after #1392 was merged and it seems to be working now. If you could verify against the MyGet build that would be helpful otherwise it should be in the next beta.

OtlpExporter now adds `peer.service` tag which makes its way to Zipkin as RemoteEndpoint and the UI shows as expected.

There's some detail here about the resolution logic: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk_exporters/zipkin.md#remote-endpoint

Closing issue but feel free to reopen if needed.

**Comment by dgrozenok** [comment]
It works for me. Thank you. I will open a separate issue to request the ServiceName option added to the OtlpExporter similar to Zipkin and Jaeger exporters to avoid the .SetResource() call with the hard-coded key "service.name".

**Comment by CodeBlanch** [comment]
@dgrozenok Sounds good regarding the OtlpExporter options issue!

Just FYI, you can do `SetResource(Resources.CreateServiceResource("service name here"))` which removes the need to know about the "service.name" key. I've been meaning to open a PR to improve that resource API just haven't had the time yet 😄

**Comment by dgrozenok** [comment]
Thanks for the suggestion! It certainly looks much better.

Added the feature request https://github.com/open-telemetry/opentelemetry-dotnet/issues/1416
